### PR TITLE
Add a component for creating baremetal hosts

### PR DESCRIPTION
**Pull request by honza**
cc @knowncitizen @jtomasek @flofuchs

### CODE REVIEW

File: src/components/CreateBaremetalHost/CreateBaremetalHost.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    });
+  };
+
+  onSubmit = async () => {
+    this.setState(state => ({
+      ...state,
+      isSubmitting: true,
+    }));
+    const data = this.state.form.value;
+    const namespace = this.props.selectedNamespace.metadata.name;
```
Comment (suomiy) [inline src/components/CreateBaremetalHost/CreateBaremetalHost.js]: you can use `getName` from selectors

Comment (honza) [reply src/components/CreateBaremetalHost/CreateBaremetalHost.js]: :heart_eyes:

File: src/k8s/request.js [hunk 2]
```diff
@@ -504,3 +512,10 @@ const getDefaultInterface = (vm, template) => {
   const defaultInterfaceName = getTemplateAnnotations(template, ANNOTATION_DEFAULT_NETWORK);
   return getDevice(vm, 'interfaces', defaultInterfaceName);
 };
+
+export const createBaremetalHost = async (k8sCreate, data) => {
```
[the remark this answers is not in the document]
Comment (honza) [reply src/k8s/request.js]: Done

File: src/components/CreateBaremetalHost/CreateBaremetalHost.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    const secretName = `${data.name.value}-secret`;
+
+    const secret = {
```
[the remark this answers is not in the document]
Comment (honza) [reply src/components/CreateBaremetalHost/CreateBaremetalHost.js]: Done

File: src/components/CreateBaremetalHost/CreateBaremetalHost.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+  render() {
+    return (
+      <Modal show bsSize="large" onHide={this.props.onClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Add Host</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Col sm={12}>
+            <p style={{ marginTop: '20px' }}>
```
[the remark this answers is not in the document]
Comment (honza) [reply src/components/CreateBaremetalHost/CreateBaremetalHost.js]: Done

**Comment by honza** [comment]
Great points, thanks

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+              fields={formFields}
+              fieldsValues={this.state.form.value}
+              onFormChange={this.onFormChange}
+              horizontal={false}
+            />
+          </Col>
+        </Modal.Body>
+        <Modal.Footer>
+          {this.state.errorMessage && <Alert type="error">{this.state.errorMessage}</Alert>}
+          {this.state.isSubmitting && <span className="spinner spinner-xs spinner-inline" />}
```
Comment (suomiy) [inline src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: please use InlineLoading component and customize it or create new one to your liking to include this use-case

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  },
+  password: {
+    id: 'password',
+    title: 'Password',
+    type: 'password',
+    required: true,
+  },
+};
+
+const formText = `Specify the host's name, management controller IP address,
```
Comment (suomiy) [inline src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: nit: please create `strings.js` file for that and use `UPPER_CASE` constant convention.

File: src/k8s/requests/hosts/index.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  data: {
+    username,
+    password,
+  },
+  type: 'Opaque',
+});
+
+const getBaremetalHostObject = (name, namespace, controller, secretName) => ({
+  apiVersion: 'metalkube.org/v1alpha1',
```
Comment (rawagner) [inline src/k8s/requests/hosts/index.js]: you should get these values (apiVersion and kind) from BaremetalHostModel so when something in model is updated (version) we will have just one place to edit in web-ui and web-ui-components. Same applies for Secret above

Comment (honza) [reply src/k8s/requests/hosts/index.js]: Done

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+  };
+
+  render() {
+    return (
+      <Modal show bsSize="large" onHide={this.props.onClose} className="kubevirt-create-baremetal-host-dialog">
+        <Modal.Header closeButton>
+          <Modal.Title>Add Host</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Col sm={12}>
```
Comment (rawagner) [inline src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: what is the reason for `Col` ?

Comment (honza) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: It makes the paragraph text at the top of the form to be aligned with the form fields; it receives some margin on the left.

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js (hunk 6 above)
Comment (honza) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: Done

**Comment by knowncitizen** [comment]
Nice

**Pull request merged**